### PR TITLE
qe_test_coverage for BZ:1670397

### PR DESCRIPTION
**Pull request by akhil-jha**
When the user is created manually(i.e., onthefly disabled) from an authsource, the attributes such as first name, last name and email should be synced with LDAP.

### CODE REVIEW

File: tests/foreman/ui/test_ldap_authentication.py [hunk 1]
```diff
@@ -1401,3 +1401,59 @@ def test_timeout_and_cac_card_ejection():
...
+    :BZ: 1670397
+
+    :expectedresults: The attributes should be synced.
+    """
+    ldap_auth_name = gen_string('alphanumeric')
+    auth_source_name = 'LDAP-' + ldap_auth_name
+    with session:
+        session.ldapauthentication.create(
+            {
+                'ldap_server.name': ldap_auth_name,
```
Comment (mirekdlugosz) [inline tests/foreman/ui/test_ldap_authentication.py:1428]: I don't see "onthefly" being disabled here. Looks like this test depends on this particular setting default value, and I can't say I'm happy about this.

Could you set that option explicitly here?

File: tests/foreman/ui/test_ldap_authentication.py [hunk 2]
```diff
@@ -1401,3 +1401,59 @@ def test_timeout_and_cac_card_ejection():
...
+    ) as ldapsession:
+        with raises(NavigationTriesExceeded) as error:
+            ldapsession.user.search('')
+        assert error.typename == "NavigationTriesExceeded"
+    with session:
+        user_values = session.user.read(ldap_data['ldap_user_name'])
+        assert user_values['user']['login'] == ldap_data['ldap_user_name']
+        assert user_values['user']['firstname'] is not None
+        assert user_values['user']['lastname'] is not None
+        assert user_values['user']['mail'] is not None
```
Comment (mirekdlugosz) [inline tests/foreman/ui/test_ldap_authentication.py:1459]: Do we have control over these values? Could we use stronger assertion than "there is some value"?

File: tests/foreman/ui/test_ldap_authentication.py (hunk 1 above)
Comment (akhil-jha) [reply tests/foreman/ui/test_ldap_authentication.py:1376]: onthefly is disabled by default, anyway, I mentioned it explicitly.

File: tests/foreman/ui/test_ldap_authentication.py (hunk 2 above)
Comment (akhil-jha) [reply tests/foreman/ui/test_ldap_authentication.py]: I have updated the code. Earlier the last name was not present in the AD. I have updated it.

**Review by mirekdlugosz — APPROVED** [review]
Code looks OK, thanks for including changes from comments.
Please add test result in comment.

**Comment by akhil-jha** [comment]
The bug is still open(New state), the test will be skipped.
```
pytest tests/foreman/ui/test_ldap_authentication.py -k test_verify_attribute_of_users_are_updated
Test session starts (platform: linux, Python 3.8.5, pytest 4.6.3, pytest-sugar 0.9.4)
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/akjha/satelliteqe/robottelo
plugins: mock-1.10.4, xdist-1.34.0, services-1.3.1, forked-1.3.0, sugar-0.9.4, cov-2.10.1
collecting ... 2020-10-06 12:43:40 - conftest - DEBUG - Collected 33 test cases

 tests/foreman/ui/test_ldap_authentication.py s                                         100% ██████████
========================================= warnings summary ============================================
Results (0.17s):
       1 skipped
      32 deselected
```

**Comment by mirekdlugosz** [comment]
> The bug is still open(New state), the test will be skipped.

Ah, right, there's `skip_if_open` call. Sorry for the noise.

**Review by rplevka — APPROVED** [review]
I'm going to ack this, but I have some comments - 
- the bug doesn't seem to be bound to UI so I would proably use API for the test. It would take fraction of the time the UI test will take and provide much more consistent reults (there are occassional errors caused by slow UI, etc.)
- I'd move the part where you create the auth source to the fixture as it is not really a part of the test (this is probably more of a formality)
- you could provide the test results with the bz decorator removed. At least we could see the test has no other issue and fails at the assertion as expected

**Pull request merged**